react-components/src/hooks/useClickedNode.tsx

```diff
@@ -54,58 +59,85 @@ export const useClickedNodeData = (): ClickedNodeData | undefined => {
     cadIntersection?.treeIndex
   ).data;
 
+  const assetMappingResult = useAssetMappingForTreeIndex(
+    cadIntersection?.model.modelId,
+    cadIntersection?.model.revisionId,
+    cadIntersection?.treeIndex
+  ).data;
+
+  return useCombinedClickedNodeData(nodeDataPromises, assetMappingResult, cadIntersection);
```

Comment:
Current behaviour is that when you click, the return data for asset mapping is first undefined and then it updates, should we update the return data only when we got the data?

Reply:
Also, if you click on "non-mapped" part of the model it will still return the data two times instead of just one. Which will trigger any dependant useEffect hook.

Reply:
The reason I update both when clicking, and then when getting data, is that it enables the user to interact with the `intersection` itself, without having to wait for the data.

Reply:
It works like this for the FDM part as well.

I did make a small change to make sure you don't get two updates when clicking an unmapped node, though :+1: